khelo/lib/ui/flow/tournament/detail/members/tournament_detail_members_screen.dart

```diff
@@ -0,0 +1,274 @@
+            notifier.removeTournamentMember(tournament.id, member);
+          },
+        ),
+      );
+    }
+
+    if (isOwner) {
+      if (currentUserId == member.id) {
+        actions.add(
+          BottomSheetAction(
+            title: context.l10n.common_transfer_ownership,
+            onTap: () async {
+              context.pop();
+              final newOwner = await SearchUserBottomSheet.show<UserModel>(
+                context,
+                emptyScreenTitle: context.l10n.common_transfer_ownership,
+                emptyScreenDescription:
+                    context.l10n.team_detail_transfer_ownership_description,
+              );
+              if (context.mounted && newOwner != null) {
+                notifier.changeTournamentOwner(tournament.id, member, newOwner);
+                context.pop();
+              }
+            },
+          ),
+        );
+      } else if (member.role == TournamentMemberRole.admin) {
+        actions.add(
+          BottomSheetAction(
+            title: context.l10n.tournament_members_make_organizer,
+            onTap: () async {
+              context.pop();
+              notifier.updateMemberRole(
+                  tournament.id, member, TournamentMemberRole.organizer);
+            },
+          ),
+        );
+        actions.add(
+          BottomSheetAction(
+            title: context.l10n.tournament_members_remove_member,
+            onTap: () async {
+              context.pop();
+              notifier.removeTournamentMember(tournament.id, member);
+            },
+          ),
+        );
+      } else {
+        actions.add(
+          BottomSheetAction(
+            title: context.l10n.common_make_admin,
+            onTap: () async {
+              context.pop();
+              notifier.updateMemberRole(
+                  tournament.id, member, TournamentMemberRole.admin);
+            },
+          ),
+        );
+        actions.add(
+          BottomSheetAction(
+            title: context.l10n.tournament_members_remove_member,
+            onTap: () async {
+              context.pop();
+              notifier.removeTournamentMember(tournament.id, member);
+            },
+          ),
+        );
+      }
+    } else if (currentUserRole == TournamentMemberRole.organizer) {
+      if (member.role != TournamentMemberRole.admin) {
+        actions.add(
+          BottomSheetAction(
+            title: context.l10n.common_make_admin,
+            onTap: () async {
+              context.pop();
+              notifier.updateMemberRole(
+                  tournament.id, member, TournamentMemberRole.admin);
+            },
+          ),
+        );
+        actions.add(
+          BottomSheetAction(
+            title: context.l10n.tournament_members_remove_member,
+            onTap: () async {
+              context.pop();
+              notifier.removeTournamentMember(tournament.id, member);
+            },
+          ),
+        );
+      }
+    }
```

Comment:
instead of this, it's better to create enum for each action and handle visible options with that.